content/guides/detect-retransmits.md

```diff
@@ -0,0 +1,45 @@
+---
+Title:       Detect Retransmits
+Author:      Sven Trittler
+Date:        2024-10-18
+Template:    plain_markdown
+---
+
+# Detecting and Optimizing DDS Retransmissions with Wireshark
+
+## Understanding DDS Retransmissions
+
+DDS (Data Distribution Service) relies on UDP, meaning messages can be lost due to network conditions or QoS settings. With reliable QoS, lost messages must be retransmitted, which can introduce latency spikes. Retransmissions can only be detected through heartbeats, which are sent either at regular intervals or with every data message. These heartbeats signal which sequence numbers have been successfully received and indicate any missing messages that need to be retransmitted.
+
+## Detecting Retransmissions in Wireshark
+
+DDS retransmissions are not explicitly marked in the data stream. However, retransmissions are triggered by a NACK (Negative Acknowledgment). To identify NACKs in Wireshark, you can use the following filter `rtps.bitmap.num_bits > 0`.
+This filter allows you to focus on NACKs and, in most cases, is sufficient for detecting retransmits.
+
+You may also filter by submessage IDs:
```

Comment:
Just looking for messages with a non-empty bitmap is indeed the first step. You can get `GAP` messages that way as well, but that is pretty rare and so suggesting to filter by submessage ID is as a potential extra step is a good one. I would add something like "e.g., using `rtps.sm.id == 0x6`" for completeness.

Reply:
i added the hint :)